File: app/src/main/java/ru/gsench/githubusers/domain/github_repo/GitHubRequests.java
package ru.gsench.githubusers.domain.github_repo;

import java.net.MalformedURLException;
import java.net.URL;

/**
 * Created by grish on 01.05.2017.
 */

public class GitHubRequests {

    public static URL searchUser(String searchFor, int limit, int offset){
        try {
            return new URL("https://api.github.com/search/users?q="+searchFor+"&page="+(offset/limit+1)+"&per_page="+limit);
        } catch (MalformedURLException e) {
            e.printStackTrace();
            return null;
        }
    }

}

File: app/src/main/java/ru/gsench/githubusers/domain/github_repo/GitHubRequests.kt
package ru.gsench.githubusers.domain.github_repo

import java.io.UnsupportedEncodingException
import java.net.MalformedURLException
import java.net.URL
import java.net.URLEncoder

/**
 * Created by grish on 01.05.2017.
 */

object GitHubRequests {

    fun searchUser(searchFor: String, limit: Int, offset: Int): URL? {
        try {
            return URL("https://api.github.com/search/users?q=" + URLEncoder.encode(searchFor, "utf-8") + "&page=" + (offset / limit + 1) + "&per_page=" + limit)
        } catch (e: MalformedURLException) {
            e.printStackTrace()
            return null
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
            return null
        }

    }

}
